R/schema.R

```diff
@@ -0,0 +1,301 @@
+#'
+#' @details Methods are not available for all DBMS.
+#'
+#' An error is thrown if no schema of that name exists.
+#'
+#' Additional arguments are:
+#'
+#'   - `dbname`: supported for MSSQL. Look for tables on a different
+#'   database on the connected MSSQL-server; default: database addressed by `dest`.
+#'
+#' @return A tibble with the following columns:
+#'   \describe{
+#'     \item{`table_name`}{name of the table,}
+#'     \item{`remote_name`}{identifier of the table on the DBMS.
+#'     Can be used to access the listed tables with the syntax
+#'     `tbl(dest, remote_name).`}
+#'   }
+#'
+#' @family schema handling functions
+#' @export
+sql_schema_table_list <- function(dest, schema = NULL, ...) {
+  if (!is_null(schema)) {
+    check_param_class(schema, "character")
+    check_param_length(schema)
+  }
+  if (!is_null(schema) && !sql_schema_exists(dest, schema, ...)) {abort_no_schema_exists(schema, ...)}
+  UseMethod("sql_schema_table_list")
+}
+
+#' @export
+`sql_schema_table_list.src_Microsoft SQL Server` <- function(dest, schema = NULL, dbname = NULL, ...) {
+  if (!is_null(dbname)) {
+    check_param_class(dbname, "character")
+    check_param_length(dbname)
+  }
+  enframe(
+    get_src_tbl_names(dest, schema = schema, dbname = dbname),
+    name = "table_name",
+    value = "remote_name") %>%
+    mutate(remote_name = dbplyr::ident_q(remote_name))
+}
+
+#' @export
+sql_schema_table_list.src_PqConnection <- function(dest, schema = NULL, ...) {
+  enframe(
+    get_src_tbl_names(dest, schema = schema),
+    name = "table_name",
+    value = "remote_name") %>%
+    mutate(remote_name = dbplyr::ident_q(remote_name))
+}
+
+#' @export
+`sql_schema_table_list.Microsoft SQL Server` <- function(dest, schema = NULL, dbname = NULL, ...) {
+  `sql_schema_table_list.src_Microsoft SQL Server`(
+    dest = dbplyr::src_dbi(dest),
+    schema = schema,
+    dbname = dbname,
+    ...)
+}
+
+#' @export
+sql_schema_table_list.PqConnection <- function(dest, schema = NULL, ...) {
+  sql_schema_table_list.src_PqConnection(
+    dest = dbplyr::src_dbi(dest),
+    schema = schema,
+    ...)
+}
+
+# sql_schema_drop() -------------------------------------------------------
+
+#' Remove a schema from a database
+#'
+#' @description `sql_schema_drop()` deletes an empty schema from the database.
+#'
+#' @inheritParams sql_schema_exists
+#'
+#' @details Methods are not available for all DBMS.
+#'
+#' An error is thrown if no schema of that name exists.
+#'
+#' Additional arguments are:
+#'
+#'   - `dbname`: supported for MSSQL. Remove a schema from a different
+#'   database on the connected MSSQL-server; default: database addressed by `dest`.
+#'
+#' @return `NULL` invisibly.
+#'
+#' @family schema handling functions
+#' @export
+sql_schema_drop <- function(dest, schema, ...) {
+  check_param_class(schema, "character")
+  check_param_length(schema)
+  if (nrow(sql_schema_table_list(dest, schema, ...)) > 0) {
+    abort_schema_not_empty(schema, ...)
```

Comment:
I wouldn't mind getting the error message straight from the database here. Some servers offer a `CASCADE` option that cleans all contents, we could match that to a `force` argument.

Reply:
yes, I was also thinking of special scenarios: maybe some DBMS allow for schemas without tables but with views, which would lead to an error when trying to drop the schema. Gonna remove this error and add the force argument.